aws-android-sdk-iot/src/androidTest/java/com/amazonaws/mobileconnectors/iot/MqttManagerIntegrationTest.java

```diff
@@ -724,37 +724,39 @@ public void onFailure(Throwable exception) {
     }
 
     @Test
-    public void mqttWebSocket() throws Exception {
-        websocketConnectionTest(IAM);
+    public void testWebSocketWithIamAuth() throws Exception {
+        websocketConnectionTest(AuthenticationMode.IAM);
     }
 
     @Test
-    public void testCustomAuthConnect() throws Exception {
-        websocketConnectionTest(CUSTOM_AUTH);
+    public void testWebSocketWithCustomAuth() throws Exception {
+        websocketConnectionTest(AuthenticationMode.CUSTOM_AUTH);
     }
 
-    private void websocketConnectionTest(AWSIotMqttManager.AuthenticationMode authMode) throws InterruptedException {
+    private void websocketConnectionTest(AuthenticationMode authMode)
+            throws InterruptedException, JSONException {
         final ArrayList<AWSIotMqttClientStatusCallback.AWSIotMqttClientStatus> statuses = new ArrayList<AWSIotMqttClientStatusCallback.AWSIotMqttClientStatus>();
         final ArrayList<String> messages = new ArrayList<String>();
 
-        AWSIotMqttManager mqttManager = new AWSIotMqttManager("int-test-w-ws", Region.getRegion(Regions.US_EAST_1), endpointPrefix);
+        AWSIotMqttManager mqttManager = new AWSIotMqttManager("int-test-w-ws",
+                Region.getRegion(Regions.US_EAST_1), endpointPrefix);
 
         mqttManager.setAutoReconnect(false);
 
-        if (authMode.equals(CUSTOM_AUTH)) {
+        if (AuthenticationMode.CUSTOM_AUTH.equals(authMode)) {
             // connect using WebSockets and custom authentication token
             mqttManager.connect(
-                    "token-header",
-                    "allow",
-                    "Gv85h71GT07RWo5uWQpqHlYWePN7YVY7KgiuPVEnt65+tDDWyZOlGLIDxvwC7LiBF7FbmG/coeBELKltqmrP0gjM7sIyXlrx315xG3LupoK4bmsNIzg6xxWa0n60WxhOv1p5qXDAhmYWIFuL0enMtuwpCY1W073NqZ/+iOsrJexssZiiLnvVf3z6HReEEER6TfFevrgAEUva0h9pzbUepbJ6RaOO7zZY6e/T/wSHrKl0jA4MalGReKbhdlg0HyTPfILhjorYkF7ew/8Ml+QMr/WU90YMp9Sv8SF0q7WBzbuqxpAR8vxm00rRlgrFdsIRjGFTDmrp9Fsx+xlx0aADCA==",
-                    "CustomAuthorizerIntegTest",
+                    getPackageConfigure().getString("token_key_name"),
```

Comment:
Most ideal would be to get this config object into a statically typed object model. That is, something like:
```
final class ConnectionOptions {
    // Can some of these things be enumerations / validated / constrained beyond just String?
    private final String tokenKeyName;
    private final String token;
    private final String tokenSignature;
    private final String customerAuthorizerName;

    ...
    static ConnectionOptions fromPackageConfig(PackageConfig config) {
        return new ConnectionOptions(
            config.getString(ConfigKey.TOKEN_KEY_NAME),
            config.getString(ConfigKey.TOKEN),
            config.getString(ConfigKey.TOKEN_SIGNATURE),
            config.getString(ConfigKey.CUSTOMER_AUTHORIZER_NAME)
        );
    }
}

enum ConfigKey {
    TOKEN_KEY_NAME("token_key_name"),
    TOKEN("token"),
    TOKEN_SIGNATURE("token_signature"),
    CUSTOM_AUTHORIZER_NAME("custom_authorizer_name")

    private final String value;

    @Override
    public String toString() {
        return value;
    }

    public static ConfigKey formString(@Nullable String value) {
        for (ConfigKey value : values()) {
            if (value.toString().equals(value)) {
               return value;
            }
        }
        throw new IllegalArgumentException("No config key with value = " + value);
    }
}
```
    


```

Reply:
I have transformed the connection params into a statically typed object. Quick note : None of the connection parameters viz tokenKeyName, token, tokenSignature and CustomAuthorizerName can be constrained in any way since these could be any string determined by the user.